tests/sparseml/pytorch/sparsification/pruning/test_modifier_pruning_acdc.py

```diff
@@ -0,0 +1,255 @@
+                modifier.scheduled_update(model, optimizer, epoch, test_steps_per_epoch)
+
+            modifier.optimizer_post_step(model, optimizer, epoch, test_steps_per_epoch)
+            if modifier._momentum_buffer_reset and isinstance(optimizer, SGD):
+                for name, param in model.named_parameters():
+                    momentum_buffer = optimizer.state[param]["momentum_buffer"]
+                    if (
+                        is_previous_phase_decompression is not None
+                        and not is_previous_phase_decompression
+                        and modifier._is_phase_decompression
+                    ):
+                        assert torch.all(momentum_buffer == 0.0).item() is True
+
+                    else:
+                        assert torch.all(momentum_buffer == 0.0).item() is False
+
+            is_previous_phase_decompression = modifier._is_phase_decompression
+            epoch += 1
+
+    def test_state_dict_save_load(
+        self,
+        modifier_lambda,
+        model_lambda,
+        optim_lambda,
+        test_steps_per_epoch,  # noqa: F811
+    ):
+        state_dict_save_load_test(
+            self,
+            modifier_lambda,
+            model_lambda,
+            optim_lambda,
+            test_steps_per_epoch,
+            False,
+        )
+
+
+def test_ac_dc_pruning_yaml():
+    compression_sparsity = 0.8
+    start_epoch = 6
+    end_epoch = 26
+    update_frequency = 1
+    params = ["re:.*weight"]
+    global_sparsity = True
+    leave_enabled = True
+    mask_type = "unstructured"
+    momentum_buffer_reset = True
+    log_types = ALL_TOKEN
+
+    yaml_str = f"""
+    !ACDCPruningModifier
+        compression_sparsity: {compression_sparsity}
+        start_epoch: {start_epoch}
+        end_epoch: {end_epoch}
+        update_frequency: {update_frequency}
+        params: {params}
+        global_sparsity: {global_sparsity}
+        leave_enabled: {leave_enabled}
+        mask_type: {mask_type}
+        momentum_buffer_reset: {momentum_buffer_reset}
+        log_types: {log_types}
+        """
+    yaml_modifier = ACDCPruningModifier.load_obj(yaml_str)  # type: ACDCPruningModifier
+    serialized_modifier = ACDCPruningModifier.load_obj(
+        str(yaml_modifier)
+    )  # type: ACDCPruningModifier
+    obj_modifier = ACDCPruningModifier(
+        compression_sparsity=compression_sparsity,
+        update_frequency=update_frequency,
+        start_epoch=start_epoch,
+        end_epoch=end_epoch,
+        params=params,
+    )
+    assert isinstance(yaml_modifier, ACDCPruningModifier)
+    assert (
+        yaml_modifier.start_epoch
+        == serialized_modifier.start_epoch
+        == obj_modifier.start_epoch
+    )
+    assert (
+        yaml_modifier.end_epoch
+        == serialized_modifier.end_epoch
+        == obj_modifier.end_epoch
+    )
+    assert yaml_modifier.params == serialized_modifier.params == obj_modifier.params
```

Comment:
need to add tests for compression sparsity, leave_enabled and others

Reply:
You mean basically a group of tests to make sure that those functionalities work properly right?